### PR TITLE
Scc 4326/adv search text fields 2

### PR DESCRIPTION
**Pull request by charmingduchess**
## Ticket:

- JIRA ticket [SCC-4326](https://newyorkpubliclibrary.atlassian.net/browse/SCC-4326)

## This PR does the following:

- Adds call number and standard numbers to advanced search text inputs
- refactor query building to rely on advanced search text input constant
- type refactoring

## Accessibility concerns or updates

-

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

[SCC-4326]: https://newyorkpubliclibrary.atlassian.net/browse/SCC-4326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Looks good but add tests.

Unrelated to your PR, the designs have the layout as 50/50 - half input elements and the second half the checkboxes. The page is actually split in thirds. First third input elements, second third checkboxes, final third is empty. Can you fix it in this PR?

File: src/utils/searchUtils.ts [hunk 1]
```diff
@@ -315,11 +321,13 @@ export function mapQueryToSearchParams({
 }: SearchQueryParams): SearchParams {
   const hasIdentifiers = issn || isbn || oclc || lccn
   const filters = collapseMultiValueQueryParams(queryFilters)
-
+  //TODO: can we merge the SearchQueryParams and SearchParams types by renaming some params? e.g. search_scope, sort, sort_direction. Also maybe passing in identifiers so it maches this pattern.
```
[the remark this answers is not in the document]
Comment (charmingduchess) [reply src/utils/searchUtils.ts:324]: That's true. I'm not really a fan of the general requirement to permanently support the original choices made in DFE. But I do understand that is the directive. It just feels like we're doing a lot of extra work in RC to accommodate it, and I wish we didn't have to.

**Review by EdwinGuzman — COMMENTED** [review]
Update looks good. Just add a few tests.

File: __test__/pages/search/advancedSearchForm.test.tsx [hunk 2]
```diff
@@ -32,32 +34,36 @@ describe("Advanced Search Form", () => {
...
-  //     // this set stimeout is to ad
-  //     // eslint-disable-next-line @typescript-eslint/no-empty-function
-  //     setTimeout(() => {}, 300)
-  //     submit()
+  //   fireEvent.change(subjectInput, { target: { value: "italian food" } })
+  //   fireEvent.change(keywordInput, { target: { value: "spaghetti" } })
+  //   fireEvent.change(contributorInput, { target: { value: "strega nonna" } })
+  //   fireEvent.change(titleInput, { target: { value: "il amore di pasta" } })
+  //   submit()
+  //   await waitFor(() =>
```
Comment (EdwinGuzman) [inline __test__/pages/search/advancedSearchForm.test.tsx:53]: Why doesn't this work?

**Review by EdwinGuzman — APPROVED** [review]
We need better test suites for this and other pages. Can you make a ticket for this test refactor and make sure to include the broken/wip test here.

File: __test__/pages/search/advancedSearchForm.test.tsx (hunk 2 above)
Comment (charmingduchess) [reply __test__/pages/search/advancedSearchForm.test.tsx:53]: I don't know! I've tried it as `fireEvent`, as `userEvents`, `userEvents` wrapped in an `act`. It has worked in the past when run alone with a `setTimeout`. I've spent so much time trying to get these to work over the past year. I'm not sure. I know it's not ideal, but it's functionality that is pretty obvious when broken, so I'm not too worried about it. I know that Emma wrote a bunch of tests for the text inputs on My Account, and I'd like to ask her to take a look at it with me.

File: src/utils/searchUtils.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { isArray, isEmpty, mapObject, forEach } from "underscore"
 
-import { textInputFields as advSearchFields } from "./advancedSearchUtils"
+import {
+  textInputFields as advSearchFields,
```
Comment (dgcohen) [inline src/utils/searchUtils.ts:4]: Does textInputFields need to be imported twice?

Comment (charmingduchess) [reply src/utils/searchUtils.ts:4]: removed!

**Review by dgcohen — APPROVED** [review]

**Pull request merged**